include/deal.II/grid/grid_generator.h

```diff
@@ -768,6 +768,64 @@ namespace GridGenerator
                       const Point<dim> &  center = Point<dim>(),
                       const double        radius = 1.);
 
+  /**
+   * Generate a 2D mesh consisting of the unit square joined with a copy shifted
+   * by $s = (1,0)$. Depending on the flags passed either the right or the left
+   * square is rotated by $\pi/2$. This way one can generate a mesh in which one
+   * square possibly contains an edge that has the opposite tangiantial (and
+   * hence also opposite normal) orientation of the neighboring edge of the
+   * other square.
+   *
+   * This mesh is not overly useful from the practical point of view. For
+   * debugging purposes it can be used to check for orientation issues for
+   * tensor valued finite elements.
+   *
+   * @note If <code>rotate_left_square==rotate_right_square</code> the mesh is consistently oriented.
+   *
+   * @tparam dim
+   * @param[out] tria input triangulation
+   * @param[in] rotate_left_square <code>true</code> if the left square is
+   * rotated by $\pi/2$
+   * @param[in] rotate_right_square <code>true</code> if the right square is
+   * rotated by $\pi/2$
```

Comment:
```suggestion
   * rotated by $\pi/2$.
```

Reply:
Actually, thinking about it, you describe a rotation by pi/2. But if you plot the orientations of edges in the reference cell, if you rotate either of the two squares by pi/2 counter-clockwise, then you still get that for both of the two squares, the edge in between points up. So you don't get a mis-oriented mesh. I think you want a rotation by pi, not pi/2.

Reply:
Yes, that is a good point. It might make for a nice follow-up, though, to extend the function to allow for all 4 rotations and extend the existing tests.